sdk/opa.go

```diff
@@ -344,6 +344,19 @@ func (opa *OPA) executeTransaction(ctx context.Context, record *server.Info, wor
 		work(s, result)
 	}
 
+	// Decision log masking only supports masking input documents that are map[string]interface{}
+	// or []interface{}. If the input was not one of those types, rebuild it from the AST.
+	switch (*record.Input).(type) {
+	case map[string]interface{}, []interface{}:
+		break
```

Comment:
[nit] This `break` statement is superfluous. If it just read
```go
	switch (*record.Input).(type) {
	case map[string]interface{}, []interface{}:
	default:
		// ...
	}
```
it would have the same effect.

Reply:
What about other native types, though? Do we need to worry about, say, `[]string`, which won't be covered by `[]interface{}` here? (Or does that _need_ the same treatment because the decision logger also can't deal with `[]string`? I can't remember 😅)

Reply:
Thanks, always having a break is something from C that I have trouble giving up.

> What about other native types, though?

The masking logic only lets you mask keys in objects. You cannot mask elements in slices (although you can reference an object key that is contained in a slice).

To your point on handling native types, I realized that checking for the type of object with the switch was not good enough, as an inner object of the `map[string]interface{}` could` be a map[string]string, for example. So I dropped the switch and am unconditionally doing the conversion. It would be nice to know if masking was enabled so we could avoid this conversion unless it is necessary, but it seems that requires a rego query to determine.